File: Pythonleetcode/Leetcode26.py
class Solution(object):
    def removeDuplicates(self, nums):
        x=[]
        for i in range(0,len(nums)):
            if i==0:
                x.append(nums[i])

            elif nums[i]!=nums[i-1]:
                x.append(nums[i])

        nums[:] =x


        return len(x)

print (Solution().removeDuplicates([1,1,2,2,2]))




# class Solution(object):
#     def removeDuplicates(self, nums):
#                 newTail=0
#                 if not nums:
#                     return 0
#
#                 index = 0
#
#                 for i in range(1, len(nums)):
#                     if nums[i] != nums[newTail]:
#                         newTail += 1
#                         nums[newTail] = nums[i]
#
#                 return newTail + 1

File: CppPractice/Project1/Project1/Leetcode26.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;


class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int ans = nums.size();
        int pointer=0;
        int temp;
        map<int, int> record;
        while (pointer< ans)
        {
            if (record[nums[pointer]]>0)
            {
                temp = nums[pointer];
                nums[pointer] = nums[ans - 1];
                nums[ans - 1] = temp;
                ans--;
            }
            else
            {
                record[nums[pointer]]++;
                pointer++;
            }

        }

        return ans;

    }
};